linkerd/distribute/src/keys.rs

```diff
@@ -0,0 +1,160 @@
+use rand::{
+    distributions::{WeightedError, WeightedIndex},
+    prelude::Distribution as _,
+    Rng,
+};
+use std::{collections::HashMap, hash::Hash};
+
+/// Uniquely identifies a key/backend pair for a distribution. This allows
+/// backends to have the same key and still participate in request distribution.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct KeyId {
+    idx: usize,
+}
+
+impl KeyId {
+    pub(crate) fn new(idx: usize) -> Self {
+        Self { idx }
+    }
+}
```

Comment:
nit: We tend to avoid mixing impls and type defs -- should generally move down below the other struct definitions.

Reply:
Done